docs/config_samples/metadata_types/bare_bone.yaml

```diff
@@ -2,10 +2,16 @@
 name: barebone
 description: A minimalist metadata type file
```

Comment:
Not saying we should fix this here, but this is so minimal that it's almost irrelevant.

Practically nobody configures metadata types, I don't think. Possibly something to consider for the future, @SpacemanPaul as the times I change a metadata type is to make a field searchable for a product... allowing arbitrary searches would be great and using metadata to configure indexed searchable fields would be 🚀

Reply:
Functionality provided by the metadata subsystem is really handy, it's a shame it can only be configured at "product creation time". Non-index based searching can be configured purely at runtime as it just about constructing a query, no need to have an index for the query to be useful. It's just that `datacube` only allows configuration from the metadata document stored in DB and linked to a given product. While "stored metadata" is handy it does not need to be the only way

Reply:
The postgis driver requires metadata type documents to be "eo3 compatible" although it's not 100% clear at the moment what that means.

The only use for metadata type documents going forwards appears to be to expose user-friendly metadata aliases for various dataset metadata entries for use in searches.